### PR TITLE
Add Location endpoint

### PR DESCRIPTION
**Pull request by orian**

### CODE REVIEW

**Comment by orian** [comment]
The complaint from codecov kind of doesn't make sense. I've put this panic("implement me") on purpose. I can remove it or drop the PR. Implementing this method requires to implement the LocationInventory endpoint.

**Comment by dbertouille** [comment]
Thanks for the PR.

Regarding the Inventory levels endpoint, my vote would be to remove it until someone gets a chance to implement it. Other services in this library only implement a subset of the Shopify endpoints. For example, the ThemeService only implements List.

If others want to leave it, it could still be tested to maintain 100% code coverage. I think something like the test below would work.

```
func TestLocationServiceOp_InventoryLevels(t *testing.T) {
	setup()
	defer teardown()

        defer func() {
            if r := recover(); r == nil {
                t.Errorf("panic expected")
            }
        }()
 	client.Location.InventoryLevels()
}
```

**Comment by orian** [comment]
Thanks. I'll drop it.

**Comment by orian** [comment]
Done.

**Comment by orian** [comment]
What about merging?

**Comment by dbertouille** [comment]
Hi @orian. My apologies, I had started a review to request a few small changes but forgot to complete it. Once those can be resolved I'd be happy to mere it.

Would you also be able to rebase to fetch the upstream changes.

Thanks!

File: location.go [hunk 1]
```diff
@@ -0,0 +1,111 @@
+package goshopify
+
+import (
+	"fmt"
+	"time"
+)
+
+const locationsBasePath = "admin/locations"
+const locationsResourceName = "locations"
```
[the remark this answers is not in the document]
Comment (orian) [reply location.go]: Done.

File: location.go [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+
+// ProductService is an interface for interfacing with the location endpoints
```
[the remark this answers is not in the document]
Comment (orian) [reply location.go]: Done.

File: location.go [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+// of the Shopify API.
+// See: https://help.shopify.com/en/api/reference/inventory/location
+type LocationService interface {
+	// Retrieves a list of locations
+	List() ([]Location, error)
```
[the remark this answers is not in the document]
Comment (orian) [reply location.go]: Done

**Pull request merged**